### ISSUE TITLE
Create fully pinned requencement file for bug-fix branch 

### ISSUE THREAD

**Issue by tacaswell** [matplotlib/matplotlib#23548]
I spent some time yesterday and today trying to get tests to run on old versions of Matplotlib which....was harder than I wished it would be.  We may want to have one very pinned requirements file (possible managed by one of the auto-bump services 😞 or tools) that we rename to `fully-pinned-v3.6.x.txt` on branching.  We can then use that (exclusively) for tests on the backport branch which should hopefully reduce the amount of dep-only backports we have to do (and provide a useful reference for anyone who wants to get old versions testing again).

_Originally posted by @tacaswell in https://github.com/matplotlib/matplotlib/issues/22574#issuecomment-1059892443_

**Comment by tacaswell** [comment]
The work here is

1. generate a proposed fully pinned version of our (maybe transient) dependencies
2. set up one of our jobs on the main branch to use it (maybe?)
3. on backport make all of the jobs in the bug-fix branch use it

I think this is a good first issue because it requires not special knowledge of Matplotlib, but does require knowledge of packaging and CI (so medium difficulty)

other references: https://zenodo.org/record/6946666 , https://peps.python.org/pep-0665/ , https://github.com/brettcannon/pip-secure-install (maybe we want to use this action?)

**Comment by navaneethkulal8** [comment]
Interested in working on #23548. Could you please guide me on how to get started with it?

**Comment by mounilKshah** [comment]
If nobody else is working on this issue, may I work on it?

**Comment by rcomer** [comment]
@mounilKshah please do.  Anyone is welcome to work on anything.  Our Contributors’ Guide is here:
https://matplotlib.org/devdocs/devel/index.html

**Comment by musharafmaqbool** [comment]
hey i would take this up?

**Comment by rcomer** [comment]
I am wondering if the uv and pixi options discussed at #30432 might actually make this redundant?

**Comment by tacaswell** [comment]
If we got a lock file checked in an tracked I think it would close this issue.

**Comment by musharafmaqbool** [comment]
hey @tacaswell 
can you review my pr and let me know if i have to make any changes?

**Comment by tacaswell** [comment]
@rcomer I misunderstood your question the first time I read it.  I thought you were referring to lock files, but I now understand to have been referring to the "as of" flags.

**Comment by rcomer** [comment]
Yes, I meant the `exclude-newer` options.  I should probably learn to be more verbose!

**Comment by patelishaan** [comment]
Hi ! I would like to work on this issue (Create fully pinned requirements file for bug-fix branches #23548) . Please assign it to me

**Comment by MuhammadArslan012** [comment]
@tacaswell 
Hi! Does this issue needs any improvement or it is closed now?
if it needs improvement please assign it to me.

**Comment by melissawm** [comment]
Hi all - we do not assign issues, so if the issue is open and does not already have an **open** PR associated with it, it is generally available for contribution. However, please try to coordinate if others have expressed interest in working on the issue recently, so you are not duplicating effort. If you have questions about the best approach for a solution, feel free to post an outline here so we can help.

**Comment by janhaviupadhay1712** [comment]
Hi, I am a new contributor and would like to work on this issue. 
Could you please assign it to me?

**Comment by gy03811-blip** [comment]
Hello, I am new to contributing to Matplotlib and exploring beginner issues. I would like to work on this issue if it is still available.

**Comment by Rahasyapandey** [comment]
Hi, @tacaswell i would like to contribute on this issue,if you could assign it to me it would be great...
For fixing the bug i think i have to read the docstring first...

**Comment by kshitiz19deshwal** [comment]
Hi, I’d like to work on this issue. Is it still available?

**Comment by melissawm** [comment]
Hi @kshitiz19deshwal , please check the (now) pinned comment:

> Hi all - we do not assign issues, so if the issue is open and does not already have an **open** PR associated with it, it is generally available for contribution. However, please try to coordinate if others have expressed interest in working on the issue recently, so you are not duplicating effort. If you have questions about the best approach for a solution, feel free to post an outline here so we can help.
